netaddr.go

```diff
@@ -323,3 +325,98 @@ func (p IPPort) TCPAddr() *net.TCPAddr {
 		Zone: zone,
 	}
 }
+
+// IPPrefix is an IP address prefix representing an IP network.
+//
+// The first Bits of IP are specified, the remaining bits match any address.
+// The range of Bits is [0,32] for IPv4 or [0,128] for IPv6.
+type IPPrefix struct {
+	IP   IP
+	Bits uint8
+}
+
+// ParseIPPrefix parses s as an IP address prefix.
+// The string can be in the form "192.168.1.0/24" or "2001::db8::/32",
+// the CIDR notation defined in RFC 4632 and RFC 4291.
+func ParseIPPrefix(s string) (IPPrefix, error) {
+	i := strings.IndexByte(s, '/')
+	if i < 0 {
+		return IPPrefix{}, fmt.Errorf("netaddr.ParseIPPrefix(%q): no '/'", s)
+	}
+	ip, err := ParseIP(s[:i])
+	if err != nil {
+		return IPPrefix{}, fmt.Errorf("netaddr.ParseIPPrefix(%q): %w", s, err)
+	}
+	s = s[i+1:]
+	bits, err := strconv.Atoi(s)
+	if err != nil {
+		return IPPrefix{}, fmt.Errorf("netaddr.ParseIPPrefix(%q): bad prefix: %w", s, err)
+	}
+	maxBits := 32
+	if ip.Is6() {
+		maxBits = 128
+	}
+	if bits < 0 || bits > maxBits {
+		return IPPrefix{}, fmt.Errorf("netaddr.ParseIPPrefix(%q): prefix length out of range", s)
+	}
+	return IPPrefix{
+		IP:   ip,
+		Bits: uint8(bits),
+	}, nil
+}
+
+// IPNet returns the net.IPNet representation of an IPPrefix.
+// The returned value is always non-nil.
+// Any zone identifier is dropped in the conversion.
+func (p IPPrefix) IPNet() *net.IPNet {
+	bits := 128
+	if p.IP.Is4() {
+		bits = 32
+	}
+	stdIP, _ := p.IP.ipZone()
+	return &net.IPNet{
+		IP:   stdIP,
+		Mask: net.CIDRMask(int(p.Bits), bits),
+	}
+}
+
+// Contains reports whether the network includes addr.
```

Comment:
"the network" here to mean the IPPrefix is a bit odd, even if if the godoc on IPPrefix calls it a network.

Maybe at least say "the network p includes addr" ?